app/models/report.rb

```diff
@@ -17,15 +17,14 @@ class Report < ApplicationRecord
   has_one :location, :through => :host
 
   validates :host_id, :status, :presence => true
-  validates :reported_at, :presence => true, :uniqueness => {:scope => [:host_id, :type]}
+  validates :reported_at, :presence => true
```

Comment:
Do you understand why this was here? This caused one test to fail:

```ruby
    def test_create_duplicate
      User.current = nil
      post :create, params: { :config_report => create_a_puppet_transaction_report }, session: set_session_user
      assert_response :success
      post :create, params: { :config_report => create_a_puppet_transaction_report }, session: set_session_user
      assert_response :unprocessable_entity
    end
```

What is the point here? I don't see any reason why the application should block uploading the same report twice, it only requires the server to search through the index. I suggest to drop this uniquness test + the index as well + the test.

Reply:
This is probably not a performance problem so I am keeping this. Let's remove this in a separate PR if needed.

Reply:
I assume the validation is there to avoid duplicate report issues. I know there were other places in the code preventing uploads too often. Then we needed to add this, because we had more types. This validation prevents ArfReport and ConfigReport creation if they were created at the same time (it happened). It's still not bulletproof, since Ansible config report can in theory appear at the same time as Puppet config report. The whole validation IMHO is useless.